servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -172,11 +172,11 @@ final class RoundRobinLoadBalancer<ResolvedAddress, C extends LoadBalancedConnec
             final int linearSearchSpace,
             @Nullable final HealthCheckConfig healthCheckConfig) {
         this.id = id + '@' + toHexString(identityHashCode(this));
-        this.targetResource = requireNonNull(targetResourceName);
-        this.eventPublisher = requireNonNull(eventPublisher);
+        this.targetResource = targetResourceName;
```

Comment:
We should keep validation for 3 arguments that come from public API: `targetResourceName`, `eventPublisher`, `connectionFactory`. Even if they are annotated as `@Nonnull`, we don't know for sure what external code passes to public methods. We can keep validation here or move it to `RoundRobinLoadBalancerFactory`, same way like you did for `DefaultLoadBalancerFactory`